### PR TITLE
Reroute Cloud Tasks to emulator when it is running

### PR DESCRIPTION
**Pull request by GarrettBurroughs**
Depends on: https://github.com/firebase/firebase-tools/pull/7475

Allows the admin SDK to enqueue and delete tasks within the locally running cloud tasks emulator to provide for easier testing. 

### API Changes

Introduces `CLOUD_TASKS_EMULATOR_HOST` which should contain the host of the Cloud Tasks emulator if it is running. When set, the admin SDK will reroute requests from `enqueue` and `delete` to the emulator rather than the production instance.

### CODE REVIEW

File: src/functions/functions-api-client-internal.ts [hunk 1]
```diff
@@ -91,7 +92,9 @@ export class FunctionsApiClient {
     }
 
     try {
-      const serviceUrl = await this.getUrl(resources, CLOUD_TASKS_API_URL_FORMAT.concat('/', id));
+      const serviceUrl = tasksEmulatorUrl(resources, functionName)?.concat('/', id) 
+        ?? await this.getUrl(resources, CLOUD_TASKS_API_URL_FORMAT.concat('/', id));
+      console.log(serviceUrl);
```
Comment (lahirumaramba) [inline src/functions/functions-api-client-internal.ts:97]: Did we mean to log the URL here?

File: src/functions/functions-api-client-internal.ts [hunk 2]
```diff
@@ -417,3 +432,10 @@ export class FirebaseFunctionsError extends PrefixedFirebaseError {
     (this as any).__proto__ = FirebaseFunctionsError.prototype;
   }
 }
+
+function tasksEmulatorUrl(resources: utils.ParsedResource, functionName: string): string | undefined {
+  if (process.env.CLOUD_TASKS_EMULATOR_HOST) {
+    return `http://${process.env.CLOUD_TASKS_EMULATOR_HOST}/projects/${resources.projectId}/locations/${resources.locationId}/queues/${functionName}/tasks`;
+  }
+  return undefined;
+}
```
Comment (lahirumaramba) [inline src/functions/functions-api-client-internal.ts:441]: nit: new line

**Review by blidd-google — COMMENTED** [review]
LGTM pending Lahiru's comments

File: src/functions/functions-api-client-internal.ts (hunk 1 above)
Comment (GarrettBurroughs) [reply src/functions/functions-api-client-internal.ts]: Nope! Just forgot to remove it.

**Review by lahirumaramba — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**